data/metric/metrics.go

```diff
@@ -85,11 +89,15 @@ func isNumeric(value interface{}) bool {
 	return err == nil
 }
 
+func castToNumeric(value interface{}) (float64, error) {
+	return strconv.ParseFloat(fmt.Sprintf("%v", value), 64)
+}
+
```

Comment:
We already discussed that this solution may affect some people that are relying in some alerts to have a gauge metric wrongly reported as any type other than float. But the solution works.

Reply:
True

I don't agree with the whole metrics api, imho we should enforce types by splitting the `SetMetric(name string, value interface{}, sourceType SourceType)` into different methods with different signatures like `SetGauge(name string, value float)`.

This change is a clear point where the user could expect some troubles.

Wdyt? @areina @MagdaPaj